db/migrate/20200325170358_add_domain_and_relative_path_to_automate.rb

```diff
@@ -0,0 +1,90 @@
+require 'ancestry'
+
+class AddDomainAndRelativePathToAutomate < ActiveRecord::Migration[5.1]
+  class MiqAeNamespace < ActiveRecord::Base
+    has_ancestry
+  end
+
+  class MiqAeClass < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    belongs_to :ae_namespace, :class_name => "MiqAeNamespace", :foreign_key => :namespace_id
+
+    def relative_path
+      "#{ae_namespace.path.pluck(:name).drop(1).join("/")}/#{name}"
+    end
+
+    def domain
+      ae_namespace.root
+    end
+  end
+
+  class MiqAeInstance < ActiveRecord::Base
+    belongs_to :ae_class, :class_name => "MiqAeClass", :foreign_key => :class_id
+  end
+
+  class MiqAeMethod < ActiveRecord::Base
+    belongs_to :ae_class, :class_name => "MiqAeClass", :foreign_key => :class_id
+  end
+
+  def up
+    add_column :miq_ae_namespaces, :domain_id, :bigint
+    add_column :miq_ae_namespaces, :domain_name, :string
+    add_column :miq_ae_namespaces, :relative_path, :string
+    add_index :miq_ae_namespaces, :domain_name
+    add_index :miq_ae_namespaces, :relative_path
+
+    add_column :miq_ae_instances, :domain_id, :bigint
+    add_column :miq_ae_instances, :domain_name, :string
+    add_column :miq_ae_instances, :relative_path, :string
+    add_index :miq_ae_instances, :domain_name
+    add_index :miq_ae_instances, :relative_path
+
+    add_column :miq_ae_methods, :domain_id, :bigint
+    add_column :miq_ae_methods, :domain_name, :string
+    add_column :miq_ae_methods, :relative_path, :string
+    add_index :miq_ae_methods, :domain_name
+    add_index :miq_ae_methods, :relative_path
+
+    say_with_time("Add domain_id, domain_name and relative_path to MiqAeNamespace") do
+      MiqAeNamespace.find_each do |ns|
+        if ns.root?
+          ns.update(:domain_name => ns.name)
+        else
+          ns.update(:domain_id => ns.root.id, :domain_name => ns.root.name, :relative_path => ns.path.pluck(:name).drop(1).join("/"))
+        end
+      end
+    end
+
+    say_with_time("Add domain_id, domain_name and relative_path to MiqAeInstance") do
+      MiqAeInstance.find_each do |i|
+        i.update(:domain_id => i.ae_class.domain.id, :domain_name => i.ae_class.domain.name, :relative_path => "#{i.ae_class.relative_path}/#{i.name}")
+      end
+    end
+
+    say_with_time("Add domain_id, domain_name and relative_path to MiqAeMethod") do
+      MiqAeMethod.find_each do |m|
+        m.update(:domain_id => m.ae_class.domain.id, :domain_name => m.ae_class.domain.name, :relative_path => "#{m.ae_class.relative_path}/#{m.name}")
+      end
```

Comment:
this is basically the same as MiqAeInstance

```suggestion
      MiqAeMethod.update_all(
        :domain_id => MiqAeNamespace.joins(:miq_ae_class).select(:domain_id).where("class_id = miq_ae_class.id"),
        :domain_name => MiqAeNamespace.joins(:miq_ae_class).select(:domain_id).where("class_id = miq_ae_class.id"),
        :relative_path => MiqAeClass.where("class_id = miq_ae_class.id").select("concat(miq_ae_class.relative_path, '/', miq_ae_method.name)")
      )
      MiqAeInstance.find_each do |i|
        i.update(:domain_id => i.ae_class.domain.id, :domain_name => i.ae_class.domain.name, :relative_path => "#{i.ae_class.relative_path}/#{i.name}")
      end
```